### PR TITLE
Kpng devyaml

### PR DESCRIPTION
**Pull request by jayunit100**
quick dev environment in hack/ to iterate against

### CODE REVIEW

File: hack/kpng-local-up.sh [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+
+	docker build -t $IMAGE ./
+	docker push $IMAGE
+
+	cd hack/
+}
+
+function install {
+	# substitute it with your changes...
+	cat kpng-deployment-ds.yaml.tmpl | sed 's/KPNG_IMAGE/$IMAGE/g' > kpng-deployment-ds.yaml
```
Comment (mcluseau) [inline hack/kpng-local-up.sh:17]: `sed "s,KPNG_IMAGE,$IMAGE,"`

File: hack/kpng-local-up.sh [hunk 2]
```diff
@@ -0,0 +1,24 @@
+#!/bin/bash
+# build the kpng image...
+
+IMAGE="jayunit100/kpng-server:latest"
+
+function build {
+	cd ../
+
+	docker build -t $IMAGE ./
+	docker push $IMAGE
```
Comment (mcluseau) [inline hack/kpng-local-up.sh:10]: isn't the push too much for a simple "local-up" ?

Comment (jayunit100) [reply hack/kpng-local-up.sh:10]: Ya figure this will evolve over Time into a make file or somethin... do u prefer me to make a make file now ?

Comment (mcluseau) [reply hack/kpng-local-up.sh:10]: I overlooked the issue, it's true that you need to push somewhere to deploy :s

File: hack/kpng-local-up.sh (hunk 1 above)
Comment (jayunit100) [reply hack/kpng-local-up.sh]: fixed

File: hack/README.md [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+
+# Contribute
+
+This is just an initial recipe for learning how kpng works.  Please contribute updates
+if you have ideas to make it better.  
+
+- One example of a starting contribution might be
+pushing directly to a local registry and configuring kind to use this reg, so dockerhub
+isnt required.  
+- Or a `tilt` recipe which hot reloads all kpng on code changes.
```
Comment (mcluseau) [inline hack/README.md:33]: I don't know `tilt` but I use modd and there's a `modd.conf`.

File: hack/kpng-deployment-ds.yaml [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+      labels:
+        app: kpng
+    spec:
+      nodeSelector:
+        kube-proxy: v2
+      serviceAccountName: kube-proxy
+      hostNetwork: true
+      # so that kpng always runs on the controlplane nodes...
+      tolerations:
+      - key: "node-role.kubernetes.io/master"
```
Comment (mcluseau) [inline hack/kpng-deployment-ds.yaml:24]: how about squeezing this long list to a single
```yaml
- effect: NoSchedule
  operator: Exists
  ```
  ?

File: hack/kpng-deployment-ds.yaml.tmpl [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+      labels:
+        app: kpng
+    spec:
+      nodeSelector:
+        kube-proxy: v2
+      serviceAccountName: kube-proxy
+      hostNetwork: true
+      # so that kpng always runs on the controlplane nodes...
+      tolerations:
+      - key: "node-role.kubernetes.io/master"
```
Comment (mcluseau) [inline hack/kpng-deployment-ds.yaml.tmpl:24]: same as above

File: hack/kpng-deployment-ds.yaml [hunk 6]
```diff
@@ -0,0 +1,86 @@
+apiVersion: apps/v1
```
Comment (mcluseau) [inline hack/kpng-deployment-ds.yaml:1]: do we want to include this file or is the `.tmpl` enough?

**Pull request merged**